bpf/lib/nodeport.h

```diff
@@ -325,7 +313,9 @@ int tail_nodeport_ipv6_dsr(struct __ctx_buff *ctx)
 		return DROP_INVALID;
 
 	fib_params.family = AF_INET6;
-	fib_params.ifindex = NATIVE_DEV_IFINDEX;
+#ifdef DIRECT_ROUTING_DEV_IFINDEX
+	fib_params.ifindex = DIRECT_ROUTING_DEV_IFINDEX;
+#endif
```

Comment:
Is this "first device in the --device list being the direct routing one" due to the above? Is the helper required to haver some ifindex as input? Could we make this more generic in that the user doesn't need to keep this in mind?

Reply:
> Is this "first device in the --device list being the direct routing one" due to the above?

Yes, currently we don't determine which iface from the --device list is used for the direct routing. In a follow-up, I will explore whether k8s Node IP addr is available before exec'ing bpf/init.sh. If yes, then we can dynamically determine such an iface, and thus the requirement for the first device in the list could be lifted.